src/Action/NullAction.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Action;
+
+/**
+ * Empty action. Useful to trigger the kernel.view event without doing anything specific in the action
+ * (e.g. the POST action).
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class NullAction
+{
+    public function __invoke()
```

Comment:
I think this is the same dilemma as above. If we zoom out and look at the bigger picture it's a bigger dilemma, which might not have been obvious to us all this while.

The `kernel.view` event is the stage where some kind of domain object (or an array of them) is being turned into a Response. When we do the deserialization in `kernel.view` we seem to have gone against that expectation (but I'm not sure if we have any better option :laughing:). FWIW `@ParamConverter` uses the `kernel.controller` event (this should be the equivalent of our data provider step).